include/deal.II/grid/grid_refinement.h

```diff
@@ -48,6 +48,41 @@ template <class T> class Vector;
 namespace GridRefinement
 {
   /**
+   * This function adjust cell number fraction of refinement and coarsening
+   * according to current cell number and maximal cell number.
+   *
+   * @arg @p current_n_cells is current cell number.
+   *
+   * @arg @p max_n_cells is specified a maximal number of cells. If current
+   * cell number is already exceeded maximal cell number, @p max_n_cells will
+   * be set to zero and @p bottom_fraction will be adjusted to reduce cell
+   * number to @ max_n_cells. If cell number is going to be exceeded only
+   * upon refinement, then refinement and coarsening fractions are going
+   * to be adjusted with a same ratio in an attempt to reach the maximum
+   * number of cells. Be aware though that through proliferation of refinement
+   * due to Triangulation::MeshSmoothing, this number is only an indicator.
+   * The default value of this argument is to impose no limit on the number
+   * of cells.
+   *
+   * @arg @p top_fraction is required refinement fraction. It's value may be
+   * changed after this function call.
```

Comment:
I think you imply that this is a fraction *of the total number of cells*, instead of a fraction *of the total error*, correct?